src/query/boolean_query/boolean_weight.rs

```diff
@@ -219,6 +219,24 @@ impl<TScoreCombiner: ScoreCombiner + Sync> Weight for BooleanWeight<TScoreCombin
         Ok(())
     }
 
+    fn for_each_no_score(
+        &self,
+        reader: &SegmentReader,
+        callback: &mut dyn FnMut(DocId),
+    ) -> crate::Result<()> {
+        let scorer = self.complex_scorer(reader, 1.0, &self.score_combiner_fn)?;
```

Comment:
Can we still use the DoNothingCombiner?

```suggestion
        let scorer = self.complex_scorer(reader, 1.0, &DoNothingCombiner)?;
```